MatomoTracker/MatomoTracker.swift

```diff
@@ -114,13 +114,14 @@ final public class MatomoTracker: NSObject {
     ///   - siteId: The unique site id generated by the server when a new site was created.
     ///   - baseURL: The url of the Matomo server. This url has to end in `piwik.php` or `matomo.php`.
     ///   - userAgent: An optional parameter for custom user agent.
-    @objc convenience public init(siteId: String, baseURL: URL, userAgent: String? = nil) {
+    ///   - useBackedQueue: An optional parameter for using UserDefaultsBackedQueue instead of the default  memory queue.
+    @objc convenience public init(siteId: String, baseURL: URL, userAgent: String? = nil, useBackedQueue: Bool = false) {
         let validSuffix = baseURL.absoluteString.hasSuffix("piwik.php") ||
             baseURL.absoluteString.hasSuffix("matomo.php")
         assert(validSuffix, "The baseURL is expected to end in piwik.php or matomo.php")
         
-        let queue = MemoryQueue()
         let dispatcher = URLSessionDispatcher(baseURL: baseURL, userAgent: userAgent)
+        let queue: Queue = useBackedQueue ? UserDefaultsBackedQueue(suiteName: "\(siteId)\(dispatcher.baseURL.absoluteString)") : MemoryQueue()
         self.init(siteId: siteId, queue: queue, dispatcher: dispatcher)
     }
```

Comment:
I think we would not even need any changes to this initializer here. The user of the SDK could use the `init(siteId: String, queue: Queue, dispatcher: Dispatcher)` directly and use the `UserDefaultsBackedQueue` as a queue there. What do you think?